### ISSUE TITLE
dotnet_style_null_propagation = true:suggestion reporting a warning 

### ISSUE THREAD

**Issue by adityapatwardhan** [dotnet/roslyn#51849]
**Version Used**: .NET 6 preview.2 SDK

**Steps to Reproduce**:

1.  Use .NET 6 preview.2 SDK
2. Have the project file with `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>`
2.  Have code like:

```csharp

return (runspacePool != null) ? (runspacePool.RemoteRunspacePoolInternal) : null;

```

3. Build the console app with runtime as linux-x64 or osx-x64 for error and win7-x64 for no error.

**Expected Behavior**: No warning emitted.

**Actual Behavior**: Warning emitted only on Linux and macOS builds not Windows

**Comment by mavasani** [comment]
I am unable to reproduce this issue. @adityapatwardhan Can you please attach a simple repro project? Also note that specifying `:severity` suffix as part of code style option specification is not recommended, instead please use the `dotnet_diagnostic.ID.severity` syntax . More details at https://github.com/dotnet/roslyn/issues/44201
